### PR TITLE
Search: Move JetpackColophon and enforce display on the free tier

### LINKED ISSUE #25906: Search Plugin: Enforce display of the “Search powered by Jetpack” on the free tier
Create a branding footer at the bottom of the overlay for free tier sites, which doesn't disappear on scrolling. 

The copy for the branding:

- If Search plugin exists regardless of whether the site has Jetpack plugin or not, we show "Powered by Jetpack Search"
- Otherwise, we show "Search powered by Jetpack"

Image

### PR DESCRIPTION
**Pull request by dognose24**
Fixes #25906 

#### Changes proposed in this Pull Request:

* Make SearchResults scrollable inside `.jetpack-instant-search__search-wrapper`
* Move JetpackColophon from Sidebar to bottom of SearchResults
* Use URL flag `free_tier=1` to show JetpackColophon temporarily

> Note: I was trying to refactor the whole SearchResults to a fixed frame with only the result list scrollable. However, users would suffer from a small monitor to see fewer results. Thus, I add the search wrapper to put the colophon at the bottom and leave the whole SearchResults scrollable.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p1HpG7-hEH-p2

#### Does this pull request change what data or activity we track or use?

N/A
#### Testing instructions:

* Navigate site url with params `/?s=&free_tier=0`
* Ensure there is **no** Jetpack colophon displaying regardless of whether the `Show "Powered by Jetpack"` in the Customizer is set or not.
截圖 2022-09-21 下午11 38 04

* Navigate site url with params `/?s=&free_tier=1`
* Ensure there is Jetpack colophon displaying regardless of whether the `Show "Powered by Jetpack"` in the Customizer is set or not.
截圖 2022-09-21 下午11 39 54

* Scroll result list on the search overlay
* Ensure the Jetpack colophon would be fixed at the same position
截圖 2022-09-21 下午11 40 03

* Click text on the Jetpack colophon
* You should be navigated to [Jetpack Search support page](https://jetpack.com/support/search/) within a new tab

### CODE REVIEW

**Comment by oskosk** [comment]
Pushed rebase in [624886f](https://github.com/Automattic/jetpack/pull/26320/commits/624886fcb33ed38c5a1c008f6452cfd287200167)

File: projects/packages/search/src/instant-search/components/search-results.jsx [hunk 1]
```diff
@@ -213,82 +214,90 @@ class SearchResults extends Component {
 	render() {
 		return (
 			<div
-				aria-hidden={ this.props.isLoading === true }
-				className="jetpack-instant-search__search-results"
+				className={ classNames( 'jetpack-instant-search__search-wrapper', {
```
Comment (jsnmoon) [inline projects/packages/search/src/instant-search/components/search-results.jsx:217]: This is a nitty suggestion that can be handled in a follow-up PR.

```suggestion
				className={ classNames( 'jetpack-instant-search__search-results-wrapper', {
```

**Review by jsnmoon — APPROVED** [review]
The overall approach looks solid, and the interface looks good. There are two regressions I recommend we tackle in a follow-up PR:

1. Prevent clicks on the colophon from closing the overlay. We'll want to update [the class name reference in the Overlay component here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/search/src/instant-search/components/overlay.jsx#L50).
2. Respect the existing colophon setting functionality. This implementation currently ignores whatever the user has configured in Customberg/Customizer and purely depends on the `free_tier` query value. I'm not current on design discussions around this — do we intend to remove the colophon configuration altogether?

File: projects/packages/search/src/instant-search/components/search-results.jsx (hunk 1 above)
Comment (dognose24) [reply projects/packages/search/src/instant-search/components/search-results.jsx:217]: Yes! I really got the uncertainty to use the `search-wrapper` or the `search-results-wrapper` as the class name. Even though I thought the latter one looked like the descendant of `search-results` in the HTML structure, it seems better with your suggestion.

Comment (dognose24) [reply projects/packages/search/src/instant-search/components/search-results.jsx:217]: I will make this update on the following fixes.

**Pull request merged**